engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -539,6 +539,8 @@ private boolean isLightVisible(Vector3f positionViewSpace, LightComponent compon
 
     @Override
     public boolean isHeadUnderWater() {
+        // TODO: Making this as a subscribable value especially for node "ChunksRefractiveReflectiveNode",
+        // TODO: glDisable and glEnable state changes on that node will be dynamically added/removed based on this value.
```

Comment:
Hmmmm... careful here. Adding/removing nodes will probably be a relatively expensive operation. The head going under/above water might happen multiple times in a short period. So, we'll have to think carefully about this. Perhaps rephrase this as a question: // TODO: review - what happens RenderGraph-wise when the head goes under water?